lib/Parser/Parse.cpp

```diff
@@ -10106,7 +10115,7 @@ ParseNodePtr Parser::ParseStatement()
     {
         if (buildAST)
         {
-            if (nullptr == m_currentNodeFunc)
+            if (nullptr == m_currentNodeFunc || IsImportOrExportStatementValidHere())
```

Comment:
Clarified 'try { return }' and others (except inside a function body) should be treated as top-level return. Fixed in 2nd iteration.

---
In reply to: [126453545](https://github.com/Microsoft/ChakraCore/pull/3307#discussion_r126453545) [](ancestors = 126453545)